### DISCUSSION TITLE
允许自定义文件输出路径

### DISCUSSION THREAD

**Discussion by WhatDamon** [taurusxin/ncmdump#25] [Ideas]
添加输出路径的支持
对于可执行文件而言可以用`--output`或者`-o`来定义，动态链接库则可以另开一个API使用
目前无法较好地实现该功能，如果使用libncmdump，则需要将输出文件进行剪切操作才可以，并且由于无法获取当前输出的文件类型为flac还是mp3，导致无法判断应该如何复制
目前的解决方案是复制ncm文件到临时目录，dump后先后尝试检测mp3和flac是否存在，然后复制到目标目录
过程过于复杂且低效，建议单独开接口（如果可以获取flac和mp3类型会更好）

**Comment by taurusxin** [comment]
对于单个文件当然可以提供输出路径，API中也会提供输出路径的选项，但是对于多文件的处理会变得复杂，因为之后有计划出个 -r 递归选项。

**Reply by blade254353074** [reply]
我理解暂时加个 --out-dir 就清晰了, input 递归应该是默认启用的

**Comment by taurusxin** [comment]
Release 1.5.0 已发布

使用方法详见
https://github.com/taurusxin/ncmdump?tab=readme-ov-file#%E5%91%BD%E4%BB%A4%E8%A1%8C%E5%B7%A5%E5%85%B7
